File: app-java/src/main/java/net/mEmoZz/yts/java/ui/main/MainContract.java
package net.mEmoZz.yts.java.ui.main;

import android.content.Context;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.RecyclerView;
import java.util.List;
import net.mEmoZz.yts.java.data.models.BaseMovie;
import net.mEmoZz.yts.java.ui.base.BasePresenter;
import net.mEmoZz.yts.java.ui.base.BaseView;

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

public interface MainContract {

  interface View extends BaseView<Presenter> {

    void setupRecycler();

    void setupRefreshLayout();

    void loadList();

    void setRecyclerAdapter(List<BaseMovie.Movie> moviesList, boolean refresh);

    void enableRefreshLayout();

    void showRefreshLayout();

    void hideRefreshLayout();

    void showLoadingBar();

    void hideLoadingBar();

    void showRecycler();

    void hideRecycler();

    void showNoConnectionView();

    void showEmptyLisView();

    void showErrorView();

    void hideHolderViews();
  }

  interface Presenter extends BasePresenter<View, MainInteractor> {

    void loadMoviesList(Context context, int pageNum, boolean refresh);

    void activateEndlessScroll(RecyclerView recyclerView, GridLayoutManager manager);
  }
}

File: app-kotlin/src/main/java/net/mEmoZz/yts/kotlin/ui/main/MainContract.kt
package net.mEmoZz.yts.kotlin.ui.main

import android.content.Context
import net.mEmoZz.yts.kotlin.data.models.BaseMovie
import net.mEmoZz.yts.kotlin.ui.base.BasePresenter
import net.mEmoZz.yts.kotlin.ui.base.BaseView

/**
 * Authored by Mohamed Fathy on 10 Mar, 2018.
 * Contact: muhamed.gendy@gmail.com
 */

interface MainContract {

  interface View : BaseView<Presenter> {

    fun setupRecycler()

    fun setupRefreshLayout()

    fun loadList()

    fun setRecyclerAdapter(moviesList: List<BaseMovie.Movie>, refresh: Boolean)

    fun enableRefreshLayout()

    fun showRefreshLayout()

    fun hideRefreshLayout()

    fun showLoadingBar()

    fun hideLoadingBar()

    fun showRecycler()

    fun hideRecycler()

    fun showNoConnectionView()

    fun showEmptyLisView()

    fun showErrorView()

    fun hideHolderViews()
  }

  interface Presenter : BasePresenter<View, MainInteractor> {

    fun loadMoviesList(context: Context, pageNum: Int, refresh: Boolean)
  }
}
